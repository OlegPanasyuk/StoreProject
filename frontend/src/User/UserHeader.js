import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Row, Col, Button } from 'react-bootstrap';
import { Link } from 'react-router-dom';


//Redux
import { connect } from 'react-redux';
import { setUserInfo } from '../REDUX/actions/actionsUser';
import { showHistoryOfShoppping } from '../REDUX/actions/actionsHistoryOfShopping';
import {
    showUserHistory,
    showUserProfile,
    showCatalogue
} from '../REDUX/actions/actionsMainContent';

// For Requests to server
import rest from 'rest';
import pathPrefix from 'rest/interceptor/pathPrefix';
import errorCode from 'rest/interceptor/errorCode';
import mime from 'rest/interceptor/mime';

const client = rest.wrap(mime, { mime: 'application/json' })
    .wrap(errorCode, { code: 500 })
    .wrap(pathPrefix, { prefix: `${process.env.REACT_APP_API_HOST}:${process.env.REACT_APP_API_PORT}` });

export class UserHeader extends Component {

    requestUserInfo() {
        let token = window.localStorage.getItem('Authorization');
        client({
            method: 'GET',
            path: '/user/',
            headers: {
                Authorization: `Bearer ${token}`
            }
        }).then(data => {
            if (data.entity === 'Unauthorized') {
                window.location.reload();
            } else {
                this.props.setUserInfo(JSON.parse(data.entity));

                this.props.showUserProfile();
            }
        });
    }

    requestUserHistoryBasket() {
        let token = window.localStorage.getItem('Authorization');
        client({
            method: 'GET',
            path: '/basket/users/history',
            headers: {
                Authorization: `Bearer ${token}`
            }
        }).then(data => {
            if (data.entity === 'Unauthorized') {
                window.location.reload();
            } else if (data.entity.length > 0) {
                this.props.showHistoryOfShoppping(data.entity);
                this.props.showUserHistory();
            }
        });
    }

    render() {
        let { userInfo, setUserInState } = this.props;
        return (

            <Row className='d-flex align-items-center justify-content-end mt-1 ml-auto'>
                <Col sm='auto' className='d-flex align-items-center'>
                    <div className='mr-1'>User: {userInfo.username}</div>
                    <Button
                        className='mr-1'
                        variant='light'
                        onClick={() => {
                            this.requestUserInfo();
                        }}>
                        <Link
                            to='/user/profile'
                        >
                            Profile
                        </Link>
                    </Button>
                    <Button
                        className='mr-1'
                        variant='light'
                        onClick={() => {
                            this.requestUserHistoryBasket();
                        }}>
                        <Link
                            to='/user/history'
                        >
                            History
                        </Link>
                    </Button>
                    <Button
                        className='mr-1'
                        variant='light'
                        onClick={() => {
                            window.localStorage.removeItem('Authorization');
                            this.props.showCatalogue();
                            setUserInState({
                                role: 'Guest'
                            });
                        }}>
                        Logout
                    </Button>
                </Col>
            </Row>

        );

    }
}

UserHeader.propTypes = {
    userInfo: PropTypes.object,
    setUserInState: PropTypes.func,
    showHistoryOfShoppping: PropTypes.func,
    showUserHistory: PropTypes.func,
    setUserInfo: PropTypes.func,
    showUserProfile: PropTypes.func,
    showCatalogue: PropTypes.func
};

const mapStoreToProps = function (state) {
    return (
        {
            historyBasket: state.userHeaderReducers.historyBasket
        }
    );
};

export default connect(mapStoreToProps, {
    showHistoryOfShoppping,
    showUserHistory,
    setUserInfo,
    showUserProfile,
    showCatalogue
})(UserHeader);

