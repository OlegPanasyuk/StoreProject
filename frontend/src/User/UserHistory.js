import React, { Component } from 'react';
import { Container } from 'react-bootstrap';
import PropTypes from 'prop-types';

//redux
import { connect } from 'react-redux';

//Simple Components
import Order from '../Order/Order';

export class UserHistory extends Component {
    render() {
        return (
            <Container>

                <h3>
                    Order History:
                </h3>
                {
                    this.props.history.map((el) => {
                        return (
                            <Order
                                key={el.id}
                                obj={el}
                            />
                        );
                    })
                }
            </Container>
        );
    }
}

UserHistory.propTypes = {
    history: PropTypes.array
};

const mapStateToProps = (state) => {
    return {
        history: state.userHeaderReducers.historyBasket
    };
};

export default connect(mapStateToProps)(UserHistory);