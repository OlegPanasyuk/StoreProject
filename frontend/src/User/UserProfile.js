import React from 'react';
import { Container } from 'react-bootstrap';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';

// Redux
import { connect } from 'react-redux';

function UserProfile({
    userInfo
}) {
    return (
        <Container>
            <h3>User Profile:</h3>
            <p>
                <b>Name:</b>
                {userInfo.username}
                <br />
                <b>Email:</b>
                {userInfo.email}
                <br />
                <b>Role:</b>
                {userInfo.role}
                <br />
                <b>Create Time:</b>
                {userInfo.create_time}
                <br />
            </p>
        </Container>
    );
}

UserProfile.propTypes = {
    userInfo: PropTypes.object
};

UserProfile.defaultProps = {
    userInfo: {}
};

const mapStateToProps = state => ({
    userInfo: state.userHeaderReducers.userInfo
});

export default withRouter(connect(mapStateToProps)(UserProfile));
