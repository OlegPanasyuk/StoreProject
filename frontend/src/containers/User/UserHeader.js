import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Row, Col, Button } from 'react-bootstrap';
import { Link } from 'react-router-dom';

// For Requests to server
import rest from 'rest';
import pathPrefix from 'rest/interceptor/pathPrefix';
import errorCode from 'rest/interceptor/errorCode';
import mime from 'rest/interceptor/mime';

// Redux
import { connect } from 'react-redux';
import { setUserInfo } from '../../actions/actionsUser';
import { showHistoryOfShoppping } from '../../actions/actionsHistoryOfShopping';
import {
    showUserHistory,
    showUserProfile,
    showCatalogue
} from '../../actions/actionsMainContent';

import UserService from '../../Services/UserService';

const client = rest.wrap(mime, { mime: 'application/json' })
    .wrap(errorCode, { code: 500 })
    .wrap(pathPrefix, { prefix: `${process.env.REACT_APP_API_HOST}:${process.env.REACT_APP_API_PORT}` });

export class UserHeader extends Component {
    requestUserInfo() {
        const token = window.localStorage.getItem('Authorization');
        const { setUserInfo, showUserProfile } = this.props;
        client({
            method: 'GET',
            path: '/user/',
            headers: {
                Authorization: `Bearer ${token}`
            }
        }).then((data) => {
            if (data.entity === 'Unauthorized') {
                window.location.reload();
            } else {
                setUserInfo(JSON.parse(data.entity));
                showUserProfile();
            }
        });
    }

    requestUserHistoryBasket() {
        const { showHistoryOfShoppping, showUserHistory } = this.props;
        UserService.getUserHistoryOfShopping()
            .then((res) => {
                if (res.data === 'Unauthorized') {
                    window.location.reload();
                } else if (res.data.length > 0) {
                    showHistoryOfShoppping(res.data);
                    showUserHistory();
                }
            });
        // client({
        //     method: 'GET',
        //     path: '/basket/users/history',
        //     headers: {
        //         Authorization: `Bearer ${token}`
        //     }
        // }).then((data) => {
        //     if (data.entity === 'Unauthorized') {
        //         window.location.reload();
        //     } else if (data.entity.length > 0) {
        //         showHistoryOfShoppping(data.entity);
        //         showUserHistory();
        //     }
        // });
    }

    render() {
        const { userInfo, setUserInState, showCatalogue } = this.props;
        return (

            <Row className='d-flex align-items-center justify-content-end mt-1 ml-auto'>
                <Col sm='auto' className='d-flex align-items-center'>
                    <div
                        className='mr-1'
                        style={{
                            color: 'var(--light)'
                        }}
                    >
                        User:
                        {userInfo.username}
                    </div>
                    <Button
                        className='mr-1'
                        variant='light'
                        onClick={() => {
                            this.requestUserInfo();
                        }}
                    >
                        <Link
                            to='/user/profile'
                        >
                            Profile
                        </Link>
                    </Button>
                    <Button
                        className='mr-1'
                        variant='light'
                        onClick={() => {
                            this.requestUserHistoryBasket();
                        }}
                    >
                        <Link
                            to='/user/history'
                        >
                            History
                        </Link>
                    </Button>
                    <Button
                        className='mr-1'
                        variant='light'
                        onClick={() => {
                            window.localStorage.removeItem('Authorization');
                            showCatalogue();
                            setUserInState({
                                role: 'Guest'
                            });
                        }}
                    >
                        Logout
                    </Button>
                </Col>
            </Row>
        );
    }
}

UserHeader.propTypes = {
    userInfo: PropTypes.object.isRequired,
    setUserInState: PropTypes.func.isRequired,
    showHistoryOfShoppping: PropTypes.func.isRequired,
    showUserHistory: PropTypes.func.isRequired,
    setUserInfo: PropTypes.func.isRequired,
    showUserProfile: PropTypes.func.isRequired,
    showCatalogue: PropTypes.func.isRequired
};

const mapStoreToProps = state => (
    {
        historyBasket: state.userHeaderReducers.historyBasket
    }
);

export default connect(mapStoreToProps, {
    showHistoryOfShoppping,
    showUserHistory,
    setUserInfo,
    showUserProfile,
    showCatalogue
})(UserHeader);
