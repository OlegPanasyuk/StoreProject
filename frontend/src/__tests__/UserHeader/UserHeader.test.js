import React from 'react';
import { UserHeader } from '../../User/UserHeader';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';
// import rest from 'rest';

configure({ adapter: new Adapter() });

// import pathPrefix from 'rest/interceptor/pathPrefix';
// import errorCode from 'rest/interceptor/errorCode';
// import mime from 'rest/interceptor/mime';

// const client = rest.wrap(mime, { mime: 'application/json' })
//     .wrap(errorCode, { code: 500 })
//     .wrap(pathPrefix, { prefix: 'http://localhost:3300' });


function setupComponent(props = {
    setUserInState: () => { },
    userInfo: {
        username: 'Admin',
        email: 'test@test.com',
        role: 'Guest'
    }
}) {
    let component = shallow(<UserHeader {...props} />);
    return { component, props };
}

// eslint-disable-next-line no-undef
describe('UserHeader', () => {
    // eslint-disable-next-line no-undef
    it('Component must be created', () => {
        const { component } = setupComponent();

        // eslint-disable-next-line no-undef
        expect(UserHeader.propTypes.userInfo).toBe(PropTypes.object);
        // eslint-disable-next-line no-undef
        expect(UserHeader.propTypes.setUserInState).toBe(PropTypes.func);

        // eslint-disable-next-line no-undef
        expect(shallowToJson(component)).toMatchSnapshot();

    });
});