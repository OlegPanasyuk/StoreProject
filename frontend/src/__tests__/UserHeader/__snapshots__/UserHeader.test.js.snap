// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserHeader Component must be created 1`] = `
<Bootstrap(Row)
  className="d-flex justify-content-end mt-1"
>
  <Col
    as="div"
    className="d-flex align-items-center"
    sm="auto"
  >
    <div
      className="mr-1"
    >
      User: 
      Admin
    </div>
    <Button
      active={false}
      className="mr-1"
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="light"
    >
      Profile
    </Button>
    <Button
      active={false}
      className="mr-1"
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="light"
    >
      History
    </Button>
    <Button
      active={false}
      className="mr-1"
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="light"
    >
      Logout
    </Button>
  </Col>
</Bootstrap(Row)>
`;
